src/charm.py

```diff
@@ -291,6 +303,98 @@ def _on_db_router_relation_changed(self, event: RelationChangedEvent) -> None:
 
         self.unit.status = ActiveStatus()
 
+    def _on_shared_db_relation_changed(self, event: RelationChangedEvent) -> None:
+        """Handle the legacy shared_db relation changed event.
+
+        Generate password and handle user and database creation for the related application.
+        """
+        if not self.unit.is_leader():
+            return
+
+        self.unit.status = MaintenanceStatus("Setting up shared-db relation")
+        logger.warning("DEPRECATION WARNING - `shared-db` is a legacy interface")
+
+        unit_relation_databag = event.relation.data[self.unit]
+        app_relation_databag = event.relation.data[self.app]
+
+        if unit_relation_databag.get("password"):
+            # Test if relation data is already set
+            # and avoid overwriting it
+            logger.warning("Data for shared-db already set.")
+            self.unit.status = ActiveStatus()
+            return
+
+        # retrieve data from the relation databag
+        # Abstracted is the fact that it is received from the leader unit of the related app
+        requires_relation_databag = event.relation.data[event.unit]
+        database_name = requires_relation_databag.get("database")
+        database_user = requires_relation_databag.get("username")
+
+        if not database_name or not database_user:
+            # Cannot create scoped database without credentials
+            logger.warning(
+                "Missing information for shared-db relation to create a database and scoped user"
+            )
+            self.unit.status = ActiveStatus()
```

Comment:
I'm never sure. My though here is, in a scenario of a running cluster with multiple apps, one should not block/wait a unit when the error is most probably come from the consumer charm.
In that manner, the unit just resume it's active status since it was previously put in Maintenance.